src/users/users.controller.ts

```diff
@@ -7,36 +7,42 @@ import {
   HttpStatus,
   Param,
   Post,
+  Res,
 } from '@nestjs/common';
 import { CreateUserDto, LoginUserDto } from '../dto/users-dto';
 import { UsersService } from './users.service';
+import { Response } from 'express';
 
 @Controller('users')
 export class UsersController {
   constructor(private readonly usersService: UsersService) {}
 
   @Post('/signup')
-  createUser(@Body() createUserDto: CreateUserDto): object {
-    const data = this.usersService.createUser(createUserDto);
-    return data;
+  async createUser(
+    @Body() createUserDto: CreateUserDto,
+    @Res() res: Response,
+  ): Promise<object> {
+    const data = await this.usersService.createUser(createUserDto);
+    const { token } = data;
+    res.header('Authorization', `Bearer ${token}`);
+    console.log(data);
+    return res.status(HttpStatus.OK).json(data);
   }
 
   @Post('/signin')
-  loginUser(@Body() loginUserDto: LoginUserDto): object {
-    console.log(loginUserDto);
-    const data = this.usersService.loginUser(loginUserDto);
-    return data;
+  async loginUser(
+    @Body() loginUserDto: LoginUserDto,
+    @Res() res: Response,
+  ): Promise<object> {
+    const data = await this.usersService.login(loginUserDto);
+    const { token } = data;
+    res.header('Authorization', `Bearer ${token}`);
+    return res.status(HttpStatus.OK).json(data);
   }
 
   @Get('/:id')
   getUserById(@Param('id') id: string): object {
-    // inspect if id with any character ex. 1a, 2k
-    if (!/^\d+$/.test(id)) {
-      throw new BadRequestException('Invalid id format');
-    }
-
-    const parsedId = parseInt(id, 10);
-    const data = this.usersService.findUserById(parsedId);
+    const data = this.usersService.findUserById(id);
```

Comment:
can just `return this.usersService.findUserById(id);` ?

Reply:
it should only return a line. 
return this.usersService.findUserById(id)